### PR TITLE
Support pam faillock with sssd enabled

### PR DESCRIPTION
**Pull request by yuumasato**
- Flexibilize check for preauth, allowing other arguments to be present.
- Flexibilize remediation so that it applies on configuration created by `authconfig`.

This fixes issues rised in https://bugzilla.redhat.com/show_bug.cgi?id=1344581#c6.

### CODE REVIEW

File: shared/templates/static/bash/accounts_passwords_pam_faillock_deny.sh [hunk 1]
```diff
@@ -5,6 +5,15 @@ populate var_accounts_passwords_pam_faillock_deny
 AUTH_FILES[0]="/etc/pam.d/system-auth"
 AUTH_FILES[1]="/etc/pam.d/password-auth"
 
+# This script fixes absence of pam_faillock.so in PAM stack or the
+# absense of deny=[0-9]+ in pam_faillock.so arguments
+# When inserting auth pam_faillock.so entries,
+# the entry with preauth argument will be added before pam_unix.so module
+# and entry with authfail argument will be added befora pam_deny.so module.
```
Comment (jan-cerny) [inline shared/templates/static/bash/accounts_passwords_pam_faillock_deny.sh]: 🌵 typo

**Comment by jan-cerny** [comment]
@yuumasato Great! LGTM. Have you checked it with SSSD authconfig on RHEL7?

**Comment by yuumasato** [comment]
@jan-cerny Yes, I have tested with 'authconfig' on RHEL7, enabling 'sssdauth' as described in bug report.

File: shared/templates/static/bash/accounts_passwords_pam_faillock_deny.sh (hunk 1 above)
Comment (yuumasato) [inline shared/templates/static/bash/accounts_passwords_pam_faillock_deny.sh]: Fixed typo.

**Comment by jan-cerny** [comment]
@yuumasato Great!

**Pull request merged**